Tasks/VsTest/vstest.ts

```diff
@@ -275,68 +310,140 @@ function executeVstest(testResultsDirectory: string, parallelRunSettingsFile: st
     return defer.promise;
 }
 
+function getVstestTestsList(vsVersion: number): Q.Promise<string> {
+    var defer = Q.defer<string>();
+    var tempFile = path.join(os.tmpdir(), uuid.v1() + ".txt");
+    tl.debug("Discovered tests listed at: " + tempFile);
+    var argsArray: string[] = [];
+
+    testAssemblyFiles.forEach(function (testAssembly) {
```

Comment:
we need to package vstest.console.exe in test selector and start to use that. this can be done in next checkin